### PR TITLE
feat(sdk): support model build sdk

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
depends on: https://github.com/star-whale/starwhale/pull/1895
features:
- support model.yaml are not in the subdir of the workdir, the following commands are ok:
    ```bash
    swcli model build . 
    swcli model build . -f /tmp/model.yaml
    swcli model build . -f another_model.yaml
    ``` 
   [asciicast](https://asciinema.org/a/565602)
- support model build SDK:  
   ```python
    from starwhale import model

    # class handler
    from .user.code.evaluator import ExamplePipelineHandler
    model.build(ExamplePipelineHandler)

    # function handler
    from .user.code.evaluator import predict_image
    model.build(predict_image)

    # module handler, @step decorates function in this module
    from .user.code import evaluator
    model.build(evaluator)

    # str handler
    model.build("user.code.evaluator:ExamplePipelineHandler")
    model.build("user.code.evaluator:predict_image")
    model.build("user.code.evaluator")
   ```
   [asciicast](https://asciinema.org/a/565603)   

## Modules
- [x] Python-SDK

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

**Review by goldenxinxing — DISMISSED** [review]
LGTM

File: client/starwhale/api/_impl/model.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        "config": [_relative(p) for p in configs if p],
+        "run": {
+            "handler": _ingest_obj_entrypoint_name(evaluator, workdir),
+        },
+        "desc": desc,
+    }
+    yaml_fpath = workdir / SW_AUTO_DIRNAME / DefaultYAMLName.MODEL
+    ensure_dir(yaml_fpath.parent)
+    ensure_file(yaml_fpath, yaml.safe_dump(config, default_flow_style=False))
+    return yaml_fpath
```
Comment (jialeicui) [inline client/starwhale/api/_impl/model.py:128]: It seems that the build function generates a config file every time. Would it be better to put it in the tmp dir (not the `.starwhale/`?

Comment (tianweidut) [reply client/starwhale/api/_impl/model.py:128]: ok

**Comment by jialeicui** [comment]
Do we need a parameter similar to "push" in this scenario where the model is automatically built in the py script?

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**